### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by freddyli7**
## Description

Add documentation for M1 and current M2 features.

### CODE REVIEW

**Review by LuaxY — APPROVED** [review]
Should be interesting to have a link to this doc from README.md to access it from main page easily

File: docs/doc.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+### Module owner
+
+#### Transaction
+
+1. Add init module owner in genesis file  
+The init module owner address and public key are required.  
+Note: this transaction is only available through CLI, and can only be executed before chain launching.
+```bash
+add-genesis-module-owner [address] [pubKey]
```
Comment (araskachoi) [inline docs/doc.md:37]:
```suggestion
chainlinkd add-genesis-module-owner [address] [pubKey]
```

File: docs/doc.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+2. Add new module owner  
+Can be signed by existing module owner only.
+```bash
+addModuleOwner [address] [pubKey]
```
Comment (araskachoi) [inline docs/doc.md:43]:
```suggestion
chainlinkd addModuleOwner [address] [pubKey]
```

File: docs/doc.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+3. Module ownership transfer  
+Can be signed by existing module owner only.   
+   The address and pubKey should match the new module owner.
+```bash
+moduleOwnershipTransfer [address] [pubKey]
```
Comment (araskachoi) [inline docs/doc.md:50]:
```suggestion
chainlinkd moduleOwnershipTransfer [address] [pubKey]
```

File: docs/doc.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+4. Add new feed  
+   Can be signed by existing module owner only.  
+   `initDataProviderList` is a string with data providers' address and pubKey connecting with comma.   
+   For example:`address1,keyKey1,address2,pubKey2`
+```bash
+addFeed [feedId] [feedOwnerAddress] [submissionCount] [heartbeatTrigger] [deviationThresholdTrigger] [initDataProviderList]
```
Comment (araskachoi) [inline docs/doc.md:58]:
```suggestion
chainlinkd addFeed [feedId] [feedOwnerAddress] [submissionCount] [heartbeatTrigger] [deviationThresholdTrigger] [initDataProviderList]
```

File: docs/doc.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+#### Query
+
+1. Get all module owners
+```bash
+getModuleOwnerList
```
Comment (araskachoi) [inline docs/doc.md:65]:
```suggestion
chainlinkd getModuleOwnerList
```

File: docs/doc.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+### Feed Owner
+
+#### Transaction
+WIP
+
+#### Query
+
+1. Get feed info by feedId
+```bash
+getFeedInfo [feedId]
```
Comment (araskachoi) [inline docs/doc.md:77]:
```suggestion
chainlinkd getFeedInfo [feedId]
```

File: docs/doc.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+### Feed Data Provider
+
+#### Transaction
+
+1. Submit feed data  
+Only valid data provider(signer of this transaction) is able to submit feed data to particular feed base on feedId.
+```bash
+submitFeedData [feedId] [feedData] [signatures]
```
Comment (araskachoi) [inline docs/doc.md:87]:
```suggestion
chainlinkd submitFeedData [feedId] [feedData] [signatures]
```

File: docs/doc.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+#### Query 
+1. Query feed data by round  
+`feedId` is optional
+```bash
+getRoundFeedData [roundId] [feedId]
```
Comment (araskachoi) [inline docs/doc.md:94]:
```suggestion
chainlinkd getRoundFeedData [roundId] [feedId]
```

File: docs/doc.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+2. Query the latest round of feed data  
+`feedId` is optional
+```bash
+getLatestFeedData [feedId]
```
Comment (araskachoi) [inline docs/doc.md:100]:
```suggestion
chainlinkd getLatestFeedData [feedId]
```

File: docs/doc.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+```bash
+getFeedInfo [feedId]
+```
+
+### Feed Data Provider
+
+#### Transaction
+
+1. Submit feed data  
+Only valid data provider(signer of this transaction) is able to submit feed data to particular feed base on feedId.
```
Comment (araskachoi) [inline docs/doc.md:85]:
```suggestion
Only a valid data provider(signer of the transaction) is able to submit feed data to a particular feed that is identified by its `feedId`.
```

File: docs/doc.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+```bash
+submitFeedData [feedId] [feedData] [signatures]
+```
+
+#### Query 
+1. Query feed data by round  
+`feedId` is optional
```
Comment (araskachoi) [inline docs/doc.md:92]:
```suggestion
`feedId` is optional; if no `feedId` is provided, all feed data for the provided round will be returned.
```

**Review by LernaJ — APPROVED** [review]

File: docs/doc.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+```bash
+getRoundFeedData [roundId] [feedId]
+```
+
+2. Query the latest round of feed data  
+`feedId` is optional
```
Comment (araskachoi) [inline docs/doc.md:98]:
```suggestion
`feedId` is optional; if no `feedId` is provided, all feed data for the latest round will be returned.
```

File: docs/doc.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+3. Module ownership transfer  
+Can be signed by existing module owner only.   
+   The address and pubKey should match the new module owner.
+```bash
+moduleOwnershipTransfer [address] [pubKey]
+```
+
+4. Add new feed  
+   Can be signed by existing module owner only.
```
Comment (araskachoi) [inline docs/doc.md:54]:
```suggestion
   Only a valid module owner is able to sign and submit this transaction.
```

File: docs/doc.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+3. Module ownership transfer  
+Can be signed by existing module owner only.   
+   The address and pubKey should match the new module owner.
+```bash
+moduleOwnershipTransfer [address] [pubKey]
+```
+
+4. Add new feed  
+   Can be signed by existing module owner only.  
+   `initDataProviderList` is a string with data providers' address and pubKey connecting with comma.
```
Comment (araskachoi) [inline docs/doc.md:55]:
```suggestion
   `initDataProviderList` is a comma-separated string with the data providers' address and pubKey.   
```

File: docs/doc.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+Can be signed by existing module owner only.   
+   The address and pubKey should match the new module owner.
+```bash
+moduleOwnershipTransfer [address] [pubKey]
+```
+
+4. Add new feed  
+   Can be signed by existing module owner only.  
+   `initDataProviderList` is a string with data providers' address and pubKey connecting with comma.   
+   For example:`address1,keyKey1,address2,pubKey2`
```
Comment (araskachoi) [inline docs/doc.md:56]:
```suggestion
   For example:`address1,pubKey1,address2,pubKey2`
```

File: docs/doc.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+2. Add new module owner  
+Can be signed by existing module owner only.
+```bash
+addModuleOwner [address] [pubKey]
+```
+
+3. Module ownership transfer  
+Can be signed by existing module owner only.
```
Comment (araskachoi) [inline docs/doc.md:47]:
```suggestion
Only a valid module owner is able to sign and submit this transaction.
```

File: docs/doc.md [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+
+1. Add init module owner in genesis file  
+The init module owner address and public key are required.  
+Note: this transaction is only available through CLI, and can only be executed before chain launching.
+```bash
+add-genesis-module-owner [address] [pubKey]
+```
+
+2. Add new module owner  
+Can be signed by existing module owner only.
```
Comment (araskachoi) [inline docs/doc.md:41]:
```suggestion
Only a valid module owner is able to sign and submit this transaction.
```

File: docs/doc.md [hunk 18]
```diff
@@ -0,0 +1,101 @@
+# Cosmos ChainLink Module Documentation
+
+Cosmos ChainLink Module is a Cosmos SDK module that allows developers to add Chainlink data feed support to their applications.
```
Comment (araskachoi) [inline docs/doc.md:3]:
```suggestion

The Cosmos ChainLink Module is a Cosmos-SDK module that allows developers to add Chainlink data feed support and off-chain reporting to their applications.
```

File: docs/doc.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+
+## Basic Concept
+
+ChainLink Module allows developers to add Chainlink data feed support to their applications with a permission control.
```
Comment (araskachoi) [inline docs/doc.md:7]:
```suggestion
The ChainLink Module allows developers to add Chainlink data feed support to their applications with some permission control. 
```

File: docs/doc.md [hunk 20]
```diff
@@ -0,0 +1,101 @@
+# Cosmos ChainLink Module Documentation
+
+Cosmos ChainLink Module is a Cosmos SDK module that allows developers to add Chainlink data feed support to their applications.
+
+## Basic Concept
+
+ChainLink Module allows developers to add Chainlink data feed support to their applications with a permission control. 
+There are three level account permission to manage the chainlink feed data:
```
Comment (araskachoi) [inline docs/doc.md:8]:
```suggestion
There are three levels of account permissions to manage in the chainlink module:
```

File: docs/doc.md [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+
+1. module owner
+2. feed owner
+3. feed data provider
+
+Module owner is a list of cosmos accounts to manage the chainlink module; however，an init module owner assigned in the genesis.json. Module owners are trusted
```
Comment (araskachoi) [inline docs/doc.md:14]:
```suggestion
The **Module Owner** is a list of cosmos accounts to manage the chainlink module. The initial Module Owner will be required to be assigned in the genesis file (`genesis.json`). Module Owners are trusted 
```

File: docs/doc.md [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+
+ChainLink Module allows developers to add Chainlink data feed support to their applications with a permission control. 
+There are three level account permission to manage the chainlink feed data:
+
+1. module owner
+2. feed owner
+3. feed data provider
+
+Module owner is a list of cosmos accounts to manage the chainlink module; however，an init module owner assigned in the genesis.json. Module owners are trusted 
+cosmos accounts which has ability to create new data feed with init feed parameters and owner of the feed. Module ownership is able to be transferred.
```
Comment (araskachoi) [inline docs/doc.md:15]:
```suggestion
cosmos accounts that has the ability to create a new `Feed` with init Feed Parameters. A Module Owner can transfer their ownership to another account.
```

File: docs/doc.md [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+
+Feed owner is a cosmos account and only one owner per a feed. The init owner of a feed is assigned by module owner when the feed is created. Feed owner is able to manage the feed parameters such as
```
Comment (araskachoi) [inline docs/doc.md:17]:
```suggestion
The **Feed Owner** is a cosmos account and there may only be one feed owner per feed. The initial feed owner of a feed is assigned by the Module Owner when the `Feed` is created. The Feed Owner is able to manage the feed parameters such as the
```

File: docs/doc.md [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+
+1. module owner
+2. feed owner
+3. feed data provider
+
+Module owner is a list of cosmos accounts to manage the chainlink module; however，an init module owner assigned in the genesis.json. Module owners are trusted 
+cosmos accounts which has ability to create new data feed with init feed parameters and owner of the feed. Module ownership is able to be transferred.
+
+Feed owner is a cosmos account and only one owner per a feed. The init owner of a feed is assigned by module owner when the feed is created. Feed owner is able to manage the feed parameters such as 
+heartbeat, feed data submission count, data providers etc. Feed ownership is able to be transferred.
```
Comment (araskachoi) [inline docs/doc.md:18]: A Feed Owner can transfer their ownership to another account.

File: docs/doc.md [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+
+Feed data provider is also a cosmos account that is able to sign the feed data submit transaction to the module. Only the valid data provider of a feed is able to submit the fee data
```
Comment (araskachoi) [inline docs/doc.md:20]:
```suggestion
A feed **Data Provider** is a cosmos account that is able to sign and submit feed data transactions to the module. Only an account that is included in the valid data provider list of a feed is able to submit any feed data
```

File: docs/doc.md [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+to that feed, valid feed data provider list is managed by the feed owner.
```
Comment (araskachoi) [inline docs/doc.md:21]:
```suggestion
to that feed. The list of valid Data Providers is managed by the Feed Owner.
```

**Review by araskachoi — CHANGES_REQUESTED** [review]

**Pull request merged**